<?xml version="1.0" ?>
<sdf version="1.6">
	<model name="dr_drift">
		<plugin filename="libvehicle_simulation_link.so" name="vehicle_simulation_link"/>
		<static>0</static>
		<allow_auto_disable>1</allow_auto_disable>
		<link name="chassis">
			<mass>1</mass>
			<visual name="visual">
				<pose frame="">0.11 0 0.06 0 0 0</pose>
				<geometry>
					<mesh>
						<uri>model://dr_drift/meshes/dr_drift.stl</uri>
						<scale>1 1 1</scale>
					</mesh>
				</geometry>
			</visual>
			<collision name="collision">
				<pose frame="">0.11 0 0.06 0 0 0</pose>
				<geometry>
					<box>
						<size>0.368 0.218 0.12</size>
					</box>
				</geometry>
			</collision>
		</link>
		<link name="camera_ros::link">
			<pose frame="">0.03 0.0 0.26 0.0 0.161 0.0</pose>
			<visual name="visual">
				<geometry>
					<box>
						<size>0.07 0.03 0.03</size>
					</box>
				</geometry>
			</visual>
			<sensor name="camera" type="camera">
				<update_rate>60</update_rate>
				<camera name="__default__">
					<horizontal_fov>2.076915850284772</horizontal_fov>
					<image>
						<width>1280</width>
						<height>1024</height>
						<format>R8G8B8</format>
					</image>
					<clip>
						<near>0.1</near>
						<far>20</far>
					</clip>
				</camera>
				<plugin filename="libgazebo_ros_camera.so" name="camera_plugin">
					<alwaysOn>1</alwaysOn>
					<updateRate>0</updateRate>
					<imageTopicName>/camera/image_uncropped_raw</imageTopicName>
					<cameraInfoTopicName>/camera/camera_info</cameraInfoTopicName>
					<frameName>vehicle</frameName>
				</plugin>
			</sensor>
		</link>
		<joint name="camera_joint" type="fixed">
			<child>camera_ros::link</child>
			<parent>chassis</parent>
		</joint>
		<link name="depth_camera_ros::link">
			<pose frame="">0.00428937 0.0114112 0.21946 0.0 0.2016281 0.0</pose>
			<visual name="visual">
				<geometry>
					<box>
						<size>0.07 0.03 0.03</size>
					</box>
				</geometry>
			</visual>
			<sensor name="camera" type="depth">
				<update_rate>60</update_rate>
				<camera name="__default__">
					<horizontal_fov>1.465630203573013</horizontal_fov>
					<image>
						<width>360</width>
						<height>240</height>
						<format>L8</format>
					</image>
					<clip>
						<near>0.1</near>
						<far>3</far>
					</clip>
				</camera>
				<plugin filename="libgazebo_ros_openni_kinect.so" name="camera_plugin">
					<alwaysOn>1</alwaysOn>
					<updateRate>0</updateRate>
					<imageTopicName>/depth_camera/raw</imageTopicName>
					<cameraInfoTopicName>/depth_camera/camera_info</cameraInfoTopicName>
					<frameName>vehicle</frameName>
				</plugin>
			</sensor>
		</link>
		<joint name="depth_camera_joint" type="fixed">
			<child>depth_camera_ros::link</child>
			<parent>chassis</parent>
		</joint>
		<link name="depth_camera_ros::link_back">
			<pose frame="">0 -0.112 0.08 0 0 -1.570796</pose>
			<visual name="visual">
				<geometry>
					<box>
						<size>0.02 0.02 0.02</size>
					</box>
				</geometry>
			</visual>
			<sensor name="camera" type="depth">
				<camera name="__default__">
					<horizontal_fov>0.436</horizontal_fov>
					<image>
						<width>10</width>
						<height>10</height>
						<format>L8</format>
					</image>
					<clip>
						<near>0.01</near>
						<far>2</far>
					</clip>
				</camera>
				<plugin filename="libgazebo_ros_openni_kinect.so" name="camera_plugin">
					<alwaysOn>1</alwaysOn>
					<updateRate>0</updateRate>
					<cameraName>tof_back</cameraName>
					<depthImageTopicName>/camera/depth_raw/back</depthImageTopicName>
					<depthImageInfoTopicName>/camera/depth_raw/back_info</depthImageInfoTopicName>
					<pointCloudTopicName>/camera/depth_raw/back_points</pointCloudTopicName>
					<frame_name>ir_back</frame_name>
					<pointCloudCutoff>0.005</pointCloudCutoff>
				</plugin>
			</sensor>
		</link>
		<joint name="depth_camera_joint_back" type="fixed">
			<child>depth_camera_ros::link_back</child>
			<parent>chassis</parent>
		</joint>
		<link name="depth_camera_ros::link_middle">
			<pose frame="">0.105 -0.1 0.08 0 0 -1.570796</pose>
			<visual name="visual">
				<geometry>
					<box>
						<size>0.02 0.02 0.02</size>
					</box>
				</geometry>
			</visual>
			<sensor name="camera" type="depth">
				<camera name="__default__">
					<horizontal_fov>0.436</horizontal_fov>
					<image>
						<width>10</width>
						<height>10</height>
						<format>L8</format>
					</image>
					<clip>
						<near>0.01</near>
						<far>2</far>
					</clip>
				</camera>
				<plugin filename="libgazebo_ros_openni_kinect.so" name="camera_plugin">
					<alwaysOn>1</alwaysOn>
					<updateRate>0</updateRate>
					<cameraName>tof_middle</cameraName>
					<depthImageTopicName>/camera/depth_raw/middle</depthImageTopicName>
					<depthImageInfoTopicName>/camera/depth_raw/middle_info</depthImageInfoTopicName>
					<pointCloudTopicName>/camera/depth_raw/middle_points</pointCloudTopicName>
					<frame_name>ir_middle</frame_name>
					<pointCloudCutoff>0.005</pointCloudCutoff>
				</plugin>
			</sensor>
		</link>
		<joint name="depth_camera_joint_middle" type="fixed">
			<child>depth_camera_ros::link_middle</child>
			<parent>chassis</parent>
		</joint>
		<link name="depth_camera_ros::link_front">
			<pose frame="">0.168 -0.1 0.08 0 0 -1.570796</pose>
			<visual name="visual">
				<geometry>
					<box>
						<size>0.02 0.02 0.02</size>
					</box>
				</geometry>
			</visual>
			<sensor name="camera" type="depth">
				<camera name="__default__">
					<horizontal_fov>0.436</horizontal_fov>
					<image>
						<width>10</width>
						<height>10</height>
						<format>L8</format>
					</image>
					<clip>
						<near>0.01</near>
						<far>2</far>
					</clip>
				</camera>
				<plugin filename="libgazebo_ros_openni_kinect.so" name="camera_plugin">
					<alwaysOn>1</alwaysOn>
					<updateRate>0</updateRate>
					<cameraName>tof_front</cameraName>
					<depthImageTopicName>/camera/depth_raw/front</depthImageTopicName>
					<depthImageInfoTopicName>/camera/depth_raw/front_info</depthImageInfoTopicName>
					<pointCloudTopicName>/camera/depth_raw/front_points</pointCloudTopicName>
					<frame_name>ir_front</frame_name>
					<pointCloudCutoff>0.005</pointCloudCutoff>
				</plugin>
			</sensor>
		</link>
		<joint name="depth_camera_joint_front" type="fixed">
			<child>depth_camera_ros::link_front</child>
			<parent>chassis</parent>
		</joint>
		<link name="depth_camera_ros::link_ahead">
			<pose frame="">0.26962 0 0.08 0 0 0</pose>
			<visual name="visual">
				<geometry>
					<box>
						<size>0.02 0.02 0.02</size>
					</box>
				</geometry>
			</visual>
			<sensor name="camera" type="depth">
				<camera name="__default__">
					<horizontal_fov>0.436</horizontal_fov>
					<image>
						<width>10</width>
						<height>10</height>
						<format>L8</format>
					</image>
					<clip>
						<near>0.01</near>
						<far>2</far>
					</clip>
				</camera>
				<plugin filename="libgazebo_ros_openni_kinect.so" name="camera_plugin">
					<alwaysOn>1</alwaysOn>
					<updateRate>0</updateRate>
					<cameraName>tof_ahead</cameraName>
					<depthImageTopicName>/camera/depth_raw/ahead</depthImageTopicName>
					<depthImageInfoTopicName>/camera/depth_raw/ahead_info</depthImageInfoTopicName>
					<pointCloudTopicName>/camera/depth_raw/ahead_points</pointCloudTopicName>
					<frame_name>ir_ahead</frame_name>
					<pointCloudCutoff>0.005</pointCloudCutoff>
				</plugin>
			</sensor>
		</link>
		<joint name="depth_camera_joint_ahead" type="fixed">
			<child>depth_camera_ros::link_ahead</child>
			<parent>chassis</parent>
		</joint>
	</model>
</sdf>
